code/modules/mining/machine_redemption.dm

```diff
@@ -165,13 +167,19 @@
 		CHECK_TICK
 
 	if(LAZYLEN(ore_buffer))
-		message_sent = FALSE
 		process_ores(ore_buffer)
-	else if(!message_sent)
-		send_console_message()
+		if(!console_notify_timer)
+			// gives 5 seconds for a load of ores to be sucked up by the ORM before it sends out request console notifications. This should be enough time for most deposits that people make
+			console_notify_timer = addtimer(CALLBACK(src, .proc/send_console_message), 50)
+
+	end_processing()
```

Comment:
By putting this here without a proper check beforehand you are actually having this cancel the process prematurely. It should only ever be doing this if it has NOTHING else on the turf its getting from to decide to stop its process. 

The proper way to check for that is actually probably just after the for loop above by checking if input has anything inside it still.

Reply:
It actually works currently because every time an ore gets dropped on the turf it calls being_processing again despite it ending prematurely. However, the way you said to handle it is probably better.